### PR TITLE
Us 936 allow to user export parameters from front

### PR DESCRIPTION
**Pull request by laurgarn**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Review by JacquesBorsenberger — APPROVED** [review]

File: src/main/java/com/powsybl/network/conversion/server/dto/ExportFormatMeta.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Class to convey an export format setting parameters descriptions
+ * @author Laurent Garnier <laurent.garnier@rte-france.com>
+ */
+@Getter
+@AllArgsConstructor
+public class ExportFormatMeta {
+
+    private final String          formatName;
```
Comment (etiennehomer) [inline src/main/java/com/powsybl/network/conversion/server/dto/ExportFormatMeta.java:22]: Done on purpose ?

File: src/main/java/com/powsybl/network/conversion/server/dto/ParamMeta.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.Getter;
+
+/**
+ * Class to convey an export format setting parameter description
+ * @author Laurent Garnier <laurent.garnier@rte-france.com>
+ */
+@Getter
+@AllArgsConstructor
+public class ParamMeta {
+    private final String        name;
```
Comment (etiennehomer) [inline src/main/java/com/powsybl/network/conversion/server/dto/ParamMeta.java:20]: same

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java [hunk 3]
```diff
@@ -235,8 +240,17 @@ ByteArrayOutputStream createZipFile(String[] listNames, MemDataSource dataSource
...
-        return Exporters.getFormats();
+    Map<String, ExportFormatMeta> getAvailableFormat() {
+        Collection<String> formatsIds = Exporters.getFormats();
+        Map<String, ExportFormatMeta> ret = formatsIds.stream().map(formatId -> {
+            Exporter exporter = Exporters.getExporter(formatId);
+            List<ParamMeta> paramsMeta = exporter.getParameters()
+                .stream().map(pp -> new ParamMeta(pp.getName(), pp.getType(), pp.getDescription(), pp.getDefaultValue()))
+                .collect(Collectors.toList());
+            return Pair.of(formatId, new ExportFormatMeta(formatId, paramsMeta));
+
```
Comment (etiennehomer) [inline src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java:251]: remove line ?

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionController.java [hunk 4]
```diff
@@ -57,28 +61,37 @@ public ResponseEntity<NetworkInfos> importCase(@Parameter(description = "Case UU
...
+        requestBody = @io.swagger.v3.oas.annotations.parameters.RequestBody(description = "Parameters for chosen format",
+            content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = Properties.class))
+        )
+    )
+    public ResponseEntity<byte[]> exportNetwork(@Parameter(description = "Network UUID") @PathVariable("mainNetworkUuid") UUID networkUuid,
                                                 @Parameter(description = "Export format")@PathVariable("format") String format,
                                                 @Parameter(description = "Variant Id") @RequestParam(name = "variantId", required = false) String variantId,
-                                                @Parameter(description = "Other networks UUID") @RequestParam(name = "networkUuid", required = false) List<String> otherNetworks
+                                                @Parameter(description = "Other networks UUID") @RequestParam(name = "networkUuid", required = false) List<String> otherNetworks,
+                                                @org.springframework.web.bind.annotation.RequestBody(required = false)
```
Comment (etiennehomer) [inline src/main/java/com/powsybl/network/conversion/server/NetworkConversionController.java:77]: On one row ?

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java (hunk 3 above)
Comment (laurgarn) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java:251]: As you wish, tough I think this (single) empty line brought some comfort in reading.

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionController.java (hunk 4 above)
Comment (laurgarn) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionController.java:77]: Done, waiting push

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**